util/packaging/docker/README.md

```diff
@@ -0,0 +1,145 @@
+Hello, world!
+```
+
+Compile and run a Chapel program using the Docker container as a
+one-shot, so you can develop outside of the container:
+```
+$ echo 'writeln("Hello, world!");' > hello.chpl
+$ docker run --rm -v "$PWD":/myapp -w /myapp chapel/chapel chpl -o hello hello.chpl
+$ docker run --rm -v "$PWD":/myapp -w /myapp chapel/chapel ./hello
+Hello, world!
+```
+
+
+## Docker Image Tags
+
+### [`chapel/chapel:<version>`](https://hub.docker.com/r/chapel/chapel)
+Supported Chapel versions:
+* [`1.28.0`, `latest`](https://github.com/chapel-lang/chapel/releases/tag/1.28.0)
+* [`1.27.0`](https://github.com/chapel-lang/chapel/releases/tag/1.27.0)
+* [`1.26.0`](https://github.com/chapel-lang/chapel/releases/tag/1.26.0)
+* [`1.25.1`](https://github.com/chapel-lang/chapel/releases/tag/1.25.1)
+* [`1.25.0`](https://github.com/chapel-lang/chapel/releases/tag/1.25.0)
+
+
+This is the core image for Chapel. It provides the complete Chapel compiler and
+runtime.  It can be used to compile and run Chapel programs inside the Docker
+container. On 64-bit Linux hosts, the compiled Chapel program binary can
+sometimes be executed outside the container (your mileage may vary).
+Other Chapel-based Docker images can be created from this image.
+
+### [`chapel/chapel-gasnet:<version>`](https://hub.docker.com/r/chapel/chapel-gasnet/)
+* `1.27.0`, `latest`
+* `1.26.0`
+* `1.25.1`
+* `1.25.0`
+
+The Chapel core image (above), rebuilt with `CHPL_COMM=gasnet` and
+`GASNET_SPAWNFN=L`. Simulates a multilocale Chapel platform within the Docker
+container.
+
+Multilocale Chapel brings additional requirements, unrelated to Docker. The
+`chpl` compilation produces two binary files (e.g. `hello_real` as well as
+`hello`). When you run the binary, you need another command line parameter,
+`-nl #`, to specify the number of locales. Please see
+[Multilocale Chapel Execution](https://chapel-lang.org/docs/usingchapel/multilocale.html)
+for details.
+
+
+## How to use the image
+
+### Compile your Chapel program
+
+The image can be used to compile and run your Chapel program inside the
+container.
+
+In the following example, you start a shell in the container and use the
+interactive session to compile and run your Chapel program, all within the
+container. The `-v` argument mounts the current directory in the container,
+`-w` sets the working directory to this mounted directory, `-i` gives you STDIN,
+and `-t` allocates a psuedo-TTY, giving you STDOUT and STDERR.
+
+The Chapel compiler, `chpl`, is invoked much like `gcc`, with argument `-o` to
+specify the name of the output binary. Then the binary file, `hello`, is run in
+the usual way.
+
+```
+$ echo 'writeln("Hello, world!");' > hello.chpl
+
+$ docker run --rm -it -v "$PWD":/myapp -w /myapp chapel/chapel /bin/bash
+root@xxxxxxxxx:/myapp# chpl -o hello hello.chpl
+root@xxxxxxxxx:/myapp# ./hello
+Hello, world!
+```
+&nbsp;
+
+Alternatively, you could compile your program in one instance of the container,
```

Comment:
Maybe say something like "referred to as one-shot" above.  Is "one-shot" a term used in the docker documentation?  Is there any chance we could avoid using a gun analogy?

Reply:
Not a Docker term as far as I know -- meant it in the sense of it just runs and exits, as opposed to entering a shell session or something like that. I see your point and will try to think of an alternative term; the best I can come up with at the moment is "one-off", but there might be something even more accurate.